client/components/SingleProduct.jsx

```diff
@@ -21,14 +66,35 @@ class SingleProduct extends Component {
           <img src={picture} alt="pic" />
           <h1>{name}</h1>
           <h3>Price: {`$` + price}</h3>
-          <ProductSelector
-            name="Quantity"
-            values={[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]}
-          />
-          <ProductSelector
-            name="Size"
-            values={[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]}
-          />
+          <FormControl
+          componentClass="select"
+          placeholder="1"
+          className="selector"
+          name='size'
+          onChange={this.handleChange}
+          >
+          {this.state.sizes.map(elem=>{
+            return(
+              <option value={elem} key={elem} name="size">
+                    {elem}
+              </option>
+            )
+          })}
+          </FormControl>
+
+          {this.state.inventoryLeft == -1 ? <h6>Select a size to see if it is available</h6>
+          : <h6>We have {this.state.inventoryLeft} left in our fancy inventory</h6>
+          }
+          {this.state.inventoryLeft == 0? <h6>OUT OF STOCK... SAD! </h6>
+          :<FormControl
+          componentClass="select"
+          placeholder="1"
+          className="selector"
+          name='size'
+          onChange={this.handleChange}
+          >
+          {this.createOptionQuantity()}
```

Comment:
maybe just do a map here, or decide to separate the other map into its own function for consistency